PyriteCore/src/world/Tools/SceneRenderTools.h

```diff
@@ -0,0 +1,161 @@
+		pyr::Camera camera;
+		pyr::GraphicalResourceRegistry registry;
+		std::function<void(const RegisteredRenderableActorCollection& sceneDescription)> renderFn;
+		struct Buffers
+		{
+			std::shared_ptr<pyr::CameraBuffer>  pcameraBuffer = std::make_shared<pyr::CameraBuffer>();
+			std::shared_ptr<ActorBuffer>		pActorBuffer = std::make_shared<ActorBuffer>();
+		} buffers;
+
+		enum RenderType { Texture2D, TextureCube };
+		DepthDrawer(RenderType type)
+		{
+			std::vector<pyr::Effect::define_t> defines{};
+		
+			if (type == TextureCube) defines.push_back(pyr::Effect::define_t{ .name = "LINEARIZE_DEPTH", .value = "1"  });
+
+			depthOnlyEffect = registry.loadEffect(
+				L"res/shaders/depthOnly.fx",
+				InputLayout::MakeLayoutFromVertex<pyr::RawMeshData::mesh_vertex_t>(),
+				defines);
+
+			renderFn = [this](const RegisteredRenderableActorCollection& sceneDescription)
+			{
+				for (const StaticMesh* smesh : sceneDescription.meshes)
+				{
+					smesh->bindModel();
+					buffers.pActorBuffer->setData(ActorBuffer::data_t{ .modelMatrix = smesh->GetTransform().getWorldMatrix() });
+					depthOnlyEffect->bindConstantBuffer("ActorBuffer", buffers.pActorBuffer);
+					depthOnlyEffect->bind();
+					std::span<const SubMesh> submeshes = smesh->getModel()->getRawMeshData()->getSubmeshes();
+					for (auto& submesh : submeshes)
+					{
+						Engine::d3dcontext().DrawIndexed(static_cast<UINT>(submesh.getIndexCount()), submesh.startIndex, 0);
+					}
+				}
+			};
+		}
+	};
+
+
+
+public:
+
+	static Texture MakeSceneDepth(const RegisteredRenderableActorCollection& sceneDescription, const Camera& camera, pyr::FrameBuffer& outFramebuffer)
+	{
+		static DepthDrawer depthDrawer2D{ DepthDrawer::Texture2D };
+
+		outFramebuffer.bind();
+		pyr::Engine::d3dcontext().IASetPrimitiveTopology(D3D11_PRIMITIVE_TOPOLOGY_TRIANGLELIST);
+		pyr::RenderProfiles::pushRasterProfile(pyr::RasterizerProfile::NOCULL_RASTERIZER);
+		pyr::RenderProfiles::pushDepthProfile(pyr::DepthProfile::TESTWRITE_DEPTH);
+
+		depthDrawer2D.buffers.pcameraBuffer->setData(pyr::CameraBuffer::data_t{
+				.mvp = camera.getViewProjectionMatrix(),
+				.pos = camera.getPosition()
+		});
+
+		depthDrawer2D.depthOnlyEffect->bindConstantBuffer("CameraBuffer", depthDrawer2D.buffers.pcameraBuffer);
+		depthDrawer2D.renderFn(sceneDescription);
+		depthDrawer2D.depthOnlyEffect->unbindResources();
+
+
+		pyr::RenderProfiles::popDepthProfile();
+		pyr::RenderProfiles::popRasterProfile();
+		outFramebuffer.unbind();
+
+		return outFramebuffer.getTargetAsTexture(pyr::FrameBuffer::DEPTH_STENCIL);
+
+	}
+
+	static Cubemap MakeSceneDepthCubemapFromPoint(const RegisteredRenderableActorCollection& sceneDescription, const vec3& worldPositon, pyr::CubemapFramebuffer& outFramebuffer)
+	{
+		static DepthDrawer depthDrawer3D{ DepthDrawer::TextureCube };
+
+		static constexpr std::array<vec3, 6> directions{
+			{
+				{1,0,0},
+				{-1,0,0},
+				{0,1,0},
+				{0,-1,0},
+				{0,0,-1},
+				{0,0, 1},
```

Comment:
that order is still strange to me, but if it works, it works

Reply:
i randomly tried stuff until it worked. 
What annoys me is that it's different from when i did IBL, which also has this kind of cubemap creation... i have no clue why they are different